### ISSUE TITLE
Structured Output

### ISSUE THREAD

**Issue by krisnova** [aurae-runtime/aurae#6]
We should be able to structure the output of aurae scripts such that it can be queried.

Sticking with the "No YAML touches our project" mentality. I propose we identify a clever way to ensure that all output from the Aurae executable is structured using valid json.

**Comment by kris-nova** [comment]
As an update right now the project is just using

```rust 
println!("{:?}", x);
```

**Comment by flostadler** [comment]
I wanted to pick this up, but I'm not fully clear on how to tackle this and would need some guidance.
Is this about users putting `print` somewhere in their scripts and this being printed as json or rather about all the output?

If it's the latter, how do you envision this working with the `raw` format? Should everything just be dumped into a json blob?
e.g.
```json
{ "raw": "RAW_MESSAGE..." }
```

What do you think about having a trait for `Output` and creating a function that ensures only types having that trait that can be registered to the Engine? It sounds a bit like a good intention because anybody could still just call `register_type_with_name` and register a type that's not serializable, but it would at least give some guardrails.

**Comment by krisnova** [comment]
Hey @flostadler welcome to the project. 

I think we have implemented a macro in AuraeScript that gives us a `.raw()` and `.json()` function for each of the auto generated types we define in the `build.rs` in AuraeScript.

I think we might be in a good place here? Although there is a good chance this code can be streamlined and improved if you wanted to take a stab at working on it.

There are examples in the AuraeScript repository that show how to use `.json()` and `.raw()`. Maybe we should change the paradigm? Maybe we should print to JSON by default? 

Unsure if we can make this more elegant for the end user writing aurae scripts?

Another thought is that we will likely need to "output" a definition for an object such that it is symmetrical to the input in an aurae script file.

In other words we should be able to "get" and object from the server, and "print" the object out in the terminal (this is what `.raw()` is supposed to do) such that it can just be copy/pasted directly into a new aurae script file.
